Src/ST-LIB_LOW/StateMachine/StateMachine.cpp

```diff
@@ -105,6 +109,13 @@ StateMachine::StateMachine(uint8_t initial_state) :
 	initial_state(initial_state), current_state(initial_state) {
 	add_state(initial_state);
 	enter_state(initial_state);
+
+	#ifdef SIM_ON
+		SharedMemory::start_emulated_state_machine();
```

Comment:
this should be started only on SharedMemory::start(), which will in turn be called on HALAL::start()